### PR TITLE
(ASC-182) qTest Uploader Prototype

### PR DESCRIPTION
**Pull request by ryan-rs**
Add a Python utility for uploading Molecule JUnitXML test results to qTest.
Note: this prototype only creates a qTest API compliant JSON file. Future
work will allow the utility to communicate directly with the qTest API.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    junit_xml = \
+    """<?xml version="1.0" encoding="utf-8"?>
+    <bad>
+    </bad>
+    """
+
+    with open(filename, 'w') as f:
+        f.write(junit_xml)
+
+    return filename
```
Comment (zreichert) [inline tests/conftest.py]: Add return character

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+To run a subset of tests::
+
+$ py.test tests.test_py_result_uploader
+
+
+Deploying
+---------
+
+A reminder for the maintainers on how to deploy.
```
Comment (johnduarte) [inline CONTRIBUTING.rst]: This line is probably not necessary.

File: HISTORY.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
+=======
+History
+=======
+
+0.1.0 (2018-03-08)
```
Comment (johnduarte) [inline HISTORY.rst]: Is this premature? I do not see such a version published on https://pypi.python.org/pypi

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+            <property name="PYTHON_ARTIFACT_SHA" value="Unknown"/>
+            <property name="APT_ARTIFACT_SHA" value="Unknown"/>
+            <property name="GIT_REPO" value="Unknown"/>
+            <property name="GIT_BRANCH" value="Unknown"/>
+        </properties>
+        <testcase classname="tests.test_default" file="tests/test_default.py" line="8" name="test_pass[ansible://localhost]" time="0.00372695922852"/>
+    </testsuite>
+    """
+
+    with open(filename, 'w') as f:
```
Comment (zreichert) [inline tests/conftest.py]: Will the decorator `@pytest.fixture(scope='session')` clean up this file once its done with it?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  - 3.4
+  - 2.7
+
+# Command to install dependencies, e.g. pip install -r requirements.txt --use-mirrors
+install: pip install -U tox-travis
+
+# Command to run tests, e.g. python setup.py test
+script: tox
+
+
```
Comment (zreichert) [inline .travis.yml]: Delete extra line

File: tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+        <testcase classname="tests.test_default" file="tests/test_default.py" line="18" name="test_pass4[ansible://localhost]" time="0.00314617156982"/>
+        <testcase classname="tests.test_default" file="tests/test_default.py" line="21" name="test_pass5[ansible://localhost]" time="0.00332307815552"/>
+    </testsuite>
+    """
+
+    with open(filename, 'w') as f:
+        f.write(junit_xml)
+
+    return filename
+
```
Comment (phongdly) [inline tests/conftest.py:189]: need an extra space here?

File: py_result_uploader/cli.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+@click.argument('json_output_file', type=click.Path())
+@click.argument('test_cycle')
+def main(json_output_file, junit_input_file, test_cycle):
+    """Upload JUnitXML results to qTest manager.
+
+    \b
+    Required Arguments:
+        JUNIT_INPUT_FILE        A valid JUnit XML results file.
+        JSON_OUTPUT_FILE        The output file to store qTest JSON test log results
+        TEST_CYCLE              The qTest cycle to use as a parent for results
```
Comment (zreichert) [inline py_result_uploader/cli.py:27]: Is the 'test cycle' concept explained in qTests documentation?

File: tests/test_py_result_uploader.py [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+        # Test
+        for x in range(len(auto_req_dict['test_logs'])):
+            for key in test_logs_exp[x]:
+                assert test_logs_exp[x][key] == auto_req_dict['test_logs'][x][key]
+
+
+class TestOutputJsonAutoRequest(object):
+    """Test cases for the 'output_json_auto_request' function"""
+
+    def test_happy_path(self, single_passing_xml, tmpdir_factory):
```
Comment (zreichert) [inline tests/test_py_result_uploader.py:198]: How does Python handle mirrored method names / param names.  In Ruby Rubocop would complain about this.

**Review by johnduarte — APPROVED** [review]

File: tests/conftest.py (hunk 6 above)
Comment (ryan-rs) [reply tests/conftest.py:189]: Yeah, it is too follow the pep8 styling standard.

**Review by zreichert — APPROVED** [review]
Tests PASS!
LGTM

**Pull request merged**